artifactory/commands/transferfiles/delayedartifactshandler.go

```diff
@@ -149,32 +149,38 @@ func handleDelayedArtifactsFiles(filesToConsume []string, base phaseBase, delayU
 		}
 		return consumeDelayedArtifactsFiles(pcWrapper, filesToConsume, uploadChunkChan, base, delayHelper, errorsChannelMng)
 	}
-	delayAction := func(pBase phaseBase, addedDelayFiles []string) error {
-		// We call this method as a recursion in order to have inner order base on the comparison function list.
-		// Remove the first delay comparison function one by one to no longer delay it until the list is empty.
-		if len(filesToConsume) > 0 && len(delayUploadComparisonFunctions) > 0 {
-			return handleDelayedArtifactsFiles(addedDelayFiles, pBase, delayUploadComparisonFunctions[1:])
-		}
-		return nil
-	}
-	return manager.doTransferWithProducerConsumer(action, delayAction)
+	return manager.doTransferWithProducerConsumer(action, nil)
 }
 
 func consumeDelayedArtifactsFiles(pcWrapper *producerConsumerWrapper, filesToConsume []string, uploadChunkChan chan UploadedChunk, base phaseBase, delayHelper delayUploadHelper, errorsChannelMng *ErrorsChannelMng) error {
+	var allDelayedArtifacts []api.FileRepresentation
+
+	// Read all delayed artifacts files
 	for _, filePath := range filesToConsume {
-		log.Debug("Handling delayed artifacts file: '" + filePath + "'")
+		log.Debug("Reading delayed artifacts file: '" + filePath + "'")
 		delayedArtifactsFile, err := readDelayFile(filePath)
 		if err != nil {
 			return err
 		}
 
-		shouldStop, err := uploadByChunks(delayedArtifactsFile.DelayedArtifacts, uploadChunkChan, base, delayHelper, errorsChannelMng, pcWrapper)
+		allDelayedArtifacts = append(allDelayedArtifacts, delayedArtifactsFile.DelayedArtifacts...)
+	}
+
+	// Upload delayed artifacts, if any exist
+	if len(allDelayedArtifacts) > 0 {
+		log.Info(fmt.Sprintf("Uploading %d delayed artifacts...", len(allDelayedArtifacts)))
+		allDelayedArtifacts = sortDelayedArtifacts(&allDelayedArtifacts, delayHelper)
+		// Use an empty delayUploadHelper to prevent further delays of these files.
+		shouldStop, err := uploadByChunks(allDelayedArtifacts, uploadChunkChan, base, delayUploadHelper{}, errorsChannelMng, pcWrapper)
```

Comment:
If a delayed artifact fails here, for example, 'conanfile.py', then the following file at the ordered delayed list is for example 'conaninfo.txt' and no issue at upload, won't it be a problem later when we try to upload It again?